dasharo-compatibility/dcu.robot

```diff
@@ -0,0 +1,160 @@
+    Skip
+    ${img_sum}=    Set Variable    f91fe017bef1f98ce292bde1c2c7c61edf7b51e9c96d25c33bfac90f50de4513
+    ${result}=    Run Process    bash    -c    ./dcu/dcu logo -l ./dcu/logo.bmp ./dcu/coreboot.rom
+    Should Contain    ${result.stdout}    Success
+    Flash Firmware    ./dcu/coreboot.rom
+
+    Power On
+    Boot Operating System    ubuntu
+    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+    Login To Linux
+    Switch To Root User
+    Write Into Terminal    sha256sum /sys/firmware/acpi/bgrt/image
+    ${out}=    Read From Terminal Until Prompt
+    ${unplugged}=    Run Keyword And Return Status
+    ...    Should Contain    ${out}    No such file
+    IF    ${unplugged} == ${TRUE}
+        Fail    Please make sure that a display device is connected to the DUT
+    END
+    Should Contain    ${out}    ${img_sum}
+
+DCU004.001 Verify SMMSTORE changes
+    [Documentation]    This test case verifies that changes made to the
+    ...    SMMSTORE via DCU are properly applied and visible in EDK2.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    DCU003.001 not supported
+
+    #
+    #    Section where the SMMSTORE gets edited via (yet unknown) DCU commands
+    #    and the resulting image gets flashed onto the DUT
+    #
+    #    We expect to set:
+    #    bool
+    #    SerialRedirection -> true
+    #    NetworkBoot -> true
+    #    uint16
+    #    Timeout -> 8
+    #    uint8
+    #    PowerFailureState -> 1
+    #    ascii
+    #    Lang -> eng
+    #    PlatformLang -> en
+    #
+
+    Power On
+
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    Should Contain Match    ${setup_menu}    *Standard English*
+
+    ${dsf_index}=    Get Index Of Matching Option In Menu    ${setup_menu}    Dasharo System Features
+    ${bmm_index}=    Get Index Of Matching Option In Menu    ${setup_menu}    Boot Maintenance Manager
+    ${rel_index}=    Evaluate    ${bmm_index}-${dsf_index}
+    ${dasharo_menu}=    Enter Dasharo System Features    ${setup_menu}
+    ${sp_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Serial Port Configuration
+    ${state}=    Get Option State    ${sp_menu}    Enable Serial Port
+    Should Be Equal    ${state}    ${TRUE}
+    Press Key N Times    2    ${ESC}
+
+    Press Key N Times And Enter    ${rel_index}    ${ARROW_DOWN}
+    ${bmm_menu}=    Get Submenu Construction
+    ${state}=    Get Option State    ${bmm_menu}    Auto Boot Time-out
+    Should Be Equal    ${state}    8
+    Press Key N Times    2    ${ESC}
+
+    Press Key N Times And Enter    ${rel_index}    ${ARROW_UP}
+    ${dasharo_menu}=    Get Submenu Construction
+    ${pmo_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Power Management Options
+    ${state}=    Get Option State    ${pmo_menu}    Power state after power
+    Should Be Equal    ${state}    Powered On
+
+
+*** Keywords ***
+Prepare DCU Test Environment
+    Set Prompt For Terminal    :~$
```

Comment:
Why such prompt?

Also `Set Prompt For Terminal` set prompts for DUT terminal, not for host.

Reply:
Removed the prompt switch completely